lib/mockingjay/strategies/perfect_tree_traversal.ex

```diff
@@ -117,42 +117,18 @@ defmodule Mockingjay.Strategies.PerfectTreeTraversal do
 
     indices = 0..(nt - 1)//2 |> Enum.into([]) |> Nx.tensor(type: :s64)
 
-    forward_args =
-      if opts[:forward] do
-        [custom_forward: opts[:forward]]
-      else
-        [
-          indices: indices,
-          num_trees: num_trees,
-          max_tree_depth: max_tree_depth,
-          features: features,
-          thresholds: thresholds,
-          root_features: root_features,
-          root_thresholds: root_thresholds,
-          values: values,
-          condition: Mockingjay.Strategy.cond_to_fun(condition),
-          n_classes: n_weak_learner_classes
-        ]
-      end
-
-    post_transform_args =
-      if opts[:post_transform] do
-        [custom_post_transform: opts[:post_transform]]
-      else
-        [n_classes: n_classes]
-      end
-
-    aggregate_args =
-      if opts[:aggregate] do
-        [custom_aggregate: opts[:aggregate]]
-      else
-        [
-          n_classes: n_classes,
-          n_trees: num_trees
-        ]
-      end
-
-    {forward_args, aggregate_args, post_transform_args}
+    [
+      indices: indices,
+      num_trees: num_trees,
+      max_tree_depth: max_tree_depth,
+      features: features,
+      thresholds: thresholds,
+      root_features: root_features,
+      root_thresholds: root_thresholds,
+      values: values,
+      condition: Mockingjay.Strategy.cond_to_fun(condition),
+      n_classes: n_weak_learner_classes
```

Comment:
```suggestion
      n_classes: n_classes
```

Reply:
I have merged it but note it was like this before. We should probably check against the Python source.

Reply:
I'm pretty sure this should he using the number of weak learner classes. That's used within forward whereas the n_classes is used after. For ensembles l, n_weak_classes will typically be 1 since the weak trees are effectively regression, whereas the overall n classes correspond to the categorical objective. The Python code calls both of these n_classes depending, and it just changes due to which class uses it

Reply:
If that's the case this PR introduced a bug, I will fix it soon.

Reply:
I have checked, this change is correct. We only use this value to shape the tensor at the end, which is used to pass how we aggregate the tensor. :)

Reply:
Ok, I see, yes I think this change is correct. I was not putting this into the context of the changes to aggregation. Thanks for checking